### PR TITLE
Fix exception when merging completion suggestions

### LINKED ISSUE #70328: Mixing field collapsing and suggestions can lead to errors
There seem to be certain circumstances when requesting completion suggestions along with a regular search that uses the "collapse" feature can lead to "array_index_out_of_bounds_exception" in the fetch phase.

I was able to reproduce the issue on master and 7.11 with the following script:

```
DELETE test

PUT test
{
  "mappings": {
    "properties": {
      "collapse_field": {
        "type": "keyword"
      },
      "suggest_field": {
        "type": "completion",
        "analyzer": "whitespace"
      }
    }
  },
  "settings": {
    "number_of_shards": 2
  }
}

POST /test/_bulk?refresh=true
{ "index" : { "_id" : 1 } }
{ "collapse_field" : "aaa", "suggest_field" : "suggestion1" }
{ "index" : { "_id" : 2 } }
{ "collapse_field" : "bbb", "suggest_field" : "suggestion2" }
{ "index" : { "_id" : 3 } }
{ "collapse_field" : "bbb", "suggest_field" : "suggestion3" }
{ "index" : { "_id" : 4 } }
{ "collapse_field" : "ccc", "suggest_field" : "suggestion4" }
{ "index" : { "_id" : 5 } }
{ "collapse_field" : "ccc", "suggest_field" : "suggestion5" }

POST /test/_search
{
  "query": {"match_all": {}}, 
  "suggest": {
    "my_suggestion": {
      "prefix": "sug",
      "completion": {
        "field": "suggest_field",
        "size": 1
      }
    }
  },
  "collapse": {
    "field": "collapse_field"
  },
  "from": 3,
  "size": 1
}
```
The request returns

```
{
  "error" : {
    "root_cause" : [ ],
    "type" : "search_phase_execution_exception",
    "reason" : "",
    "phase" : "fetch",
    "grouped" : true,
    "failed_shards" : [ ],
    "caused_by" : {
      "type" : "array_index_out_of_bounds_exception",
      "reason" : "Index 1 out of bounds for length 1"
    }
  },
  "status" : 500
}
```

My guess is this only happens with at least 2 shards, but I might be mistaken. When I omitted the document ids, reproduction sometimes worked, sometimes failed, so I assume the distribution of documents on the shards also matters.

Stack trace of the exception is coming from `SearchPhaseController.merge` where we are exceeding the array bounds:
```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at org.elasticsearch.action.search.SearchPhaseController.merge(SearchPhaseController.java:272) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.action.search.FetchSearchPhase.moveToNextPhase(FetchSearchPhase.java:217) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.action.search.FetchSearchPhase.lambda$innerRun$1(FetchSearchPhase.java:101) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.action.search.CountedCollector.countDown(CountedCollector.java:40) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.action.search.ArraySearchPhaseResults.consumeResult(ArraySearchPhaseResults.java:35) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.action.search.CountedCollector.onResult(CountedCollector.java:48) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
        at org.elasticsearch.action.search.FetchSearchPhase$2.innerOnResponse(FetchSearchPhase.java:171) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
```

### PR DESCRIPTION
**Pull request by cbuescher**
Under certain circumstances we can get "array_index_out_of_bounds" exceptions in
the fetch phase when merging comletion suggestion results with additonal field
collapsing in place. We store the ScoreDoc array in the SortedTopDocs score docs
contain both regular search hits and completion suggestion results added in
SearchPhaseController#sortDocs, so when merging these we need to know the
correct array offset where the completion results begin. This is based on the
hits length calculated in SearchPhaseController#getHits, which doesn't take into
account that there might be suggestion results present when calculating the
number of hits. This change adds the number of suggestions to SortedTopDocs in
order to be able to later account for it when calculating the hits.

Closes #70328

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

File: server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java [hunk 1]
```diff
@@ -146,11 +146,12 @@ static SortedTopDocs sortDocs(boolean ignoreFrom, final Collection<TopDocs> topD
         if (topDocs.isEmpty() && reducedCompletionSuggestions.isEmpty()) {
             return SortedTopDocs.EMPTY;
         }
+        // TODO look what we get here
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java:149]: anything left to be done here?

**Review by javanna — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java (hunk 1 above)
Comment (cbuescher) [reply server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java:149]: No, that was a bookmark while debugging this. Will remove.

**Comment by cbuescher** [comment]
@javanna thanks for the review, will merge to master and 7.13

**Comment by cbuescher** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**